src/app/services/code-view/angular-code-service.ts

```diff
@@ -216,8 +218,9 @@ export class AngularCodeService extends CodeService {
 
     private getAngularSampleMetadataUrl(demosBaseUrl: string, sampleUrl: string) {
         let demoFileMetadataName = sampleUrl.replace(demosBaseUrl + "/", "");
-        demoFileMetadataName = demoFileMetadataName.replace("/", "--");
-        let demoFileMetadataPath = `${demosBaseUrl}${this.demoFilesFolderUrlPath}${demoFileMetadataName}.json`;
+
+        let demoFileMetadataPath = this.getDemoFileSampleUrl(demoFileMetadataName, demosBaseUrl);
```

Comment:
Actually, extracting logic in `getDemoFileSampleUrl` is excessive. Current `getAngularSampleMetadataUrl` is supposed to do the same, according to names.

Move the logic from getDemoFileSampleUrl to current method, and the n you will see that `getDemoFileSampleUrl` is called replace 3 times, which I believe can be improved.